src/app/etl/ingest_external_data.py

```diff
@@ -0,0 +1,97 @@
+import re
+import sqlite3
+import polars as pl
+
+from config.settings import AppConfig
+from src.base.base_model import BaseModel
+from src.helpers.db_utils import execute_query
+from src.helpers.logger import LoggerFactory
+from src.helpers.utils import extract_report_date, get_files_in_directory, read_file_as_string
+
+logger = LoggerFactory.get_logger(__name__)
+
+class IngestFundsData(BaseModel):
+    """
+    Class for ingesting fund data from external CSV files.
+
+    This class reads fund position data from CSV files in a specified directory,
+    extracts the report date from the filename, and loads the data into the raw
+    fund position details table in the database.
+    """
+
+    def __init__(self, app_config: AppConfig):
+        """
+        Initialize the IngestFundsData class.
+
+        Args:
+            app_config (AppConfig): Application configuration object containing paths and settings
+        """
+        super().__init__(app_config)
+        self.config = app_config
+        self.db_path = self.config.db_path
+        self.funds_folder = self.config.funds_folder
```

Comment:
Returning to an earlier point, these could be Pydantic attributes associated with this class instead of being accessed via config object.